File: src/main/java/br/com/kronos/backend/aplicacao/matricula/Turma.java
package br.com.kronos.backend.aplicacao.matricula;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import br.com.kronos.backend.aplicacao.validador.Validacoes;
import lombok.Builder;
import lombok.Getter;

@Getter
public class Turma {
	
	public static final String COMBO_CACHE_DIARIO_FREQUENCIA = "combo_turmas_diario_frequencia";
	public static final String COMBO_CACHE_NAME_PERFIL_PROFESSORES = "combo_turmas_perfil_professores";
	public static final String COMBO_CACHE_NAME_ANOS_POR_CURSO = "combo_anos_turmas_por_curso";
	public static final String COMBO_CACHE_NAME_POR_CURSO_E_ANO = "combo_turmas_por_curso_e_ano";
	public static final String COMBO_CACHE_NAME_OCORRENCIAS_POR_MATRICULA = "combo_turmas_ocorrencias_por_matricula";
	
    private long id;
    private Integer ano;
    private String sigla;
    private boolean ativa;
    private boolean aberta;
    private boolean encerrada;
    private Long idPeriodoExecucao;
    private Long idCalendario;
    private Integer idTipoTurno;

	@Builder
    public Turma(long id, Integer ano, String sigla, boolean ativa, boolean aberta, boolean encerrada, Long idPeriodoExecucao, Long idCalendario,
    		Integer idTipoTurno) {

		super();
        this.id = id;
        this.ano = exigirNaoNulo(ano, "Ano");  
        this.sigla = Validacoes.exigirTamanho(sigla, "Sigla", 1, 10);
        this.ativa = ativa;
        this.aberta = aberta; 
        this.encerrada = encerrada; 
        this.idPeriodoExecucao = exigirNaoNulo(idPeriodoExecucao, "Período de execução"); 
        this.idCalendario = exigirNaoNulo(idCalendario, "Calendário"); 
        this.idTipoTurno = exigirNaoNulo(idTipoTurno, "Tipo turno"); 
	}
}

File: poc-kronos-frontend/src/model/turma/Turma.js
class Turma {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.ano = dados ? dados.ano : null;
    this.sigla = dados ? dados.sigla : null;
    this.ativa = dados ? dados.ativa : null;
    this.aberta = dados ? dados.aberta : null;
    this.encerrada = dados ? dados.encerrada : null;
    this.idPeriodoExecucao = dados ? dados.idPeriodoExecucao : null;
    this.tipoPeriodo = dados ? dados.tipoPeriodo : null;
    this.idCalendario = dados ? dados.idCalendario : null;
    this.calendario = dados ? dados.calendario : null;
    this.idTipoTurno = dados ? dados.idTipoTurno : null;
  }

  static validations(required) {
    return {
      sigla: { required },
      ano: { required },
      idPeriodoExecucao: { required },
      idCalendario: { required },
      idTipoTurno: { required }
    };
  }

  static validationMessages() {
    return {
      sigla: {
        required: "Sigla é obrigatória"
      },
      ano: {
        required: "Ano é obrigatório"
      },
      idPeriodoExecucao: {
        required: "Período Execução é obrigatória"
      },
      idCalendario: {
        required: "Calendário é obrigatório"
      },
      idTipoTurno: {
        required: "Turno é obrigatório"
      }
    };
  }
}
export default Turma;
